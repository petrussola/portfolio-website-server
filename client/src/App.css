.App {
  text-align: center;
}

a {
  text-decoration: none;
  color: inherit;
}

a:hover {
  color: #f26bac;
}

h1 {
  font-size: 3rem;
  font-weight: bold;
  padding: 1rem;
}

h2 {
  font-size: 2.5rem;
  padding: 1rem;
}

h3 {
  font-size: 2rem;
  padding: 1rem;
}

h4 {
  font-size: 1.5rem;
  padding: 1rem;
}

.bold {
  font-weight: bold;
}

p {
  line-height: 1rem;
}

.pinky {
  color: #f26bac;
}

.justify {
  text-align: justify;
}

.center {
  text-align: center;
}

.spacedlines {
  line-height: 1.6;
}

.containerMainPage {
  padding: 50px;
}

.submitButtonPage {
  border: 1px solid #A9A9A9;
}

.submitButtonFooter {
  border: none;
}
