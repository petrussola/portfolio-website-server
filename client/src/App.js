import React from "react";
import "./App.css";
import { Route } from "react-router-dom";

// IMPORT COMPONENTS
import Header from "./components/Header/Header";
import HomePage from "./components/HomePage";
import About from "./components/About/About";
import Project from "./components/ProjectPage/Project";
import Center from "./components/Header/Center";
import Footer from "./components/Footer/Footer";
import ContactPage from "./components/Contact/ContactPage";

function App() {
  return (
    <div className="App">
      <Route path="/" component={Header} />
      <Route path="/" component={Center} />
      <Route exact path="/" component={HomePage} />
      <Route path="/about" component={About} />
      <Route path="/contact" component={ContactPage} />
      <Route path="/projects/:id" component={Project} />
      <Route path="/" component={Footer} />
    </div>
  );
}

export default App;
