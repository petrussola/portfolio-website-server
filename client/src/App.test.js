import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { MemoryRouter  as Router } from 'react-router-dom';
import App from './App';

afterEach(rtl.cleanup);

let wrapper;

beforeEach(() => {
  wrapper = rtl.render(
    <Router>
      <App />
    </Router>
  );
});

it('renders without crashing', () => {
  expect(wrapper.queryByText(/Pere Sola Claver/i)).toBeInTheDocument();
});
