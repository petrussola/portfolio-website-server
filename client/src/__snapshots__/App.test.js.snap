// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="makeStyles-root-72"
    >
      <div
        class="makeStyles-root-73"
      >
        <a
          href="/"
        >
          <h1>
            P
          </h1>
        </a>
      </div>
      <div
        class="makeStyles-root-74"
      >
        <ul
          class="makeStyles-menuItems-75"
        >
          <a
            href="/"
          >
            <li>
              <h4
                class="bold"
              >
                HOME
              </h4>
            </li>
          </a>
          <a
            href="/projects/1"
          >
            <li>
              <h4
                class="bold"
              >
                PROJECTS
              </h4>
            </li>
          </a>
          <a
            href="/about"
          >
            <li>
              <h4
                class="bold"
              >
                ABOUT
              </h4>
            </li>
          </a>
          <a
            href="/contact"
          >
            <li>
              <h4
                class="bold"
              >
                CONTACT
              </h4>
            </li>
          </a>
        </ul>
      </div>
    </div>
    <div
      class="makeStyles-root-76"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-icon-77 MuiSvgIcon-fontSizeLarge"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <circle
          cx="12"
          cy="4"
          r="2"
        />
        <path
          d="M15.89 8.11C15.5 7.72 14.83 7 13.53 7h-2.54C8.24 6.99 6 4.75 6 2H4c0 3.16 2.11 5.84 5 6.71V22h2v-6h2v6h2V10.05L18.95 14l1.41-1.41-4.47-4.48z"
        />
      </svg>
      <p />
      <p>
        Pere Sola Claver
      </p>
      <p>
        /
      </p>
      <p>
        Full Stack Web Developer
      </p>
    </div>
    <div
      class="makeStyles-root-87 containerMainPage"
    >
      <ul
        class="MuiGridList-root makeStyles-gridList-88"
        style="margin: -25px;"
      >
        <li
          class="MuiGridListTile-root"
          style="width: 100%; height: 350px; padding: 25px;"
        >
          <div
            class="MuiGridListTile-tile"
          >
            <img
              alt="Tipsease - Tips made easy"
              class="makeStyles-image-91"
              src="https://i.imgur.com/QVpMzIF.png"
            />
            <div
              class="MuiGridListTileBar-root makeStyles-titleBar-89 MuiGridListTileBar-titlePositionTop"
            >
              <div
                class="MuiGridListTileBar-titleWrap MuiGridListTileBar-titleWrapActionPosLeft"
              >
                <div
                  class="MuiGridListTileBar-title"
                >
                  Tipsease - Tips made easy
                </div>
              </div>
              <div
                class="MuiGridListTileBar-actionIcon MuiGridListTileBar-actionIconActionPosLeft"
              >
                <button
                  aria-label="star Tipsease - Tips made easy"
                  class="MuiButtonBase-root MuiIconButton-root makeStyles-icon-90"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <a
                      href="/projects/1"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11.5 9C10.12 9 9 10.12 9 11.5s1.12 2.5 2.5 2.5 2.5-1.12 2.5-2.5S12.88 9 11.5 9zM20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm-3.21 14.21l-2.91-2.91c-.69.44-1.51.7-2.39.7C9.01 16 7 13.99 7 11.5S9.01 7 11.5 7 16 9.01 16 11.5c0 .88-.26 1.69-.7 2.39l2.91 2.9-1.42 1.42z"
                        />
                      </svg>
                    </a>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </li>
        <li
          class="MuiGridListTile-root"
          style="width: 66.66666666666667%; height: 650px; padding: 25px;"
        >
          <div
            class="MuiGridListTile-tile"
          >
            <img
              alt="Rick & Morty Fan Page"
              class="makeStyles-image-91"
              src="https://i.imgur.com/YFViMUy.png"
            />
            <div
              class="MuiGridListTileBar-root makeStyles-titleBar-89 MuiGridListTileBar-titlePositionTop"
            >
              <div
                class="MuiGridListTileBar-titleWrap MuiGridListTileBar-titleWrapActionPosLeft"
              >
                <div
                  class="MuiGridListTileBar-title"
                >
                  Rick & Morty Fan Page
                </div>
              </div>
              <div
                class="MuiGridListTileBar-actionIcon MuiGridListTileBar-actionIconActionPosLeft"
              >
                <button
                  aria-label="star Rick & Morty Fan Page"
                  class="MuiButtonBase-root MuiIconButton-root makeStyles-icon-90"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <a
                      href="/projects/2"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11.5 9C10.12 9 9 10.12 9 11.5s1.12 2.5 2.5 2.5 2.5-1.12 2.5-2.5S12.88 9 11.5 9zM20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm-3.21 14.21l-2.91-2.91c-.69.44-1.51.7-2.39.7C9.01 16 7 13.99 7 11.5S9.01 7 11.5 7 16 9.01 16 11.5c0 .88-.26 1.69-.7 2.39l2.91 2.9-1.42 1.42z"
                        />
                      </svg>
                    </a>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </li>
        <li
          class="MuiGridListTile-root"
          style="width: 33.333333333333336%; height: 650px; padding: 25px;"
        >
          <div
            class="MuiGridListTile-tile"
          >
            <img
              alt="Star Wars API app"
              class="makeStyles-image-91"
              src="https://i.imgur.com/NzvRHTK.png"
            />
            <div
              class="MuiGridListTileBar-root makeStyles-titleBar-89 MuiGridListTileBar-titlePositionTop"
            >
              <div
                class="MuiGridListTileBar-titleWrap MuiGridListTileBar-titleWrapActionPosLeft"
              >
                <div
                  class="MuiGridListTileBar-title"
                >
                  Star Wars API app
                </div>
              </div>
              <div
                class="MuiGridListTileBar-actionIcon MuiGridListTileBar-actionIconActionPosLeft"
              >
                <button
                  aria-label="star Star Wars API app"
                  class="MuiButtonBase-root MuiIconButton-root makeStyles-icon-90"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <a
                      href="/projects/3"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11.5 9C10.12 9 9 10.12 9 11.5s1.12 2.5 2.5 2.5 2.5-1.12 2.5-2.5S12.88 9 11.5 9zM20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm-3.21 14.21l-2.91-2.91c-.69.44-1.51.7-2.39.7C9.01 16 7 13.99 7 11.5S9.01 7 11.5 7 16 9.01 16 11.5c0 .88-.26 1.69-.7 2.39l2.91 2.9-1.42 1.42z"
                        />
                      </svg>
                    </a>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </li>
        <li
          class="MuiGridListTile-root"
          style="width: 33.333333333333336%; height: 350px; padding: 25px;"
        >
          <div
            class="MuiGridListTile-tile"
          >
            <img
              alt="Women's World Cup Football search"
              class="makeStyles-image-91"
              src="https://i.imgur.com/8rISTOo.png"
            />
            <div
              class="MuiGridListTileBar-root makeStyles-titleBar-89 MuiGridListTileBar-titlePositionTop"
            >
              <div
                class="MuiGridListTileBar-titleWrap MuiGridListTileBar-titleWrapActionPosLeft"
              >
                <div
                  class="MuiGridListTileBar-title"
                >
                  Women's World Cup Football search
                </div>
              </div>
              <div
                class="MuiGridListTileBar-actionIcon MuiGridListTileBar-actionIconActionPosLeft"
              >
                <button
                  aria-label="star Women's World Cup Football search"
                  class="MuiButtonBase-root MuiIconButton-root makeStyles-icon-90"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <a
                      href="/projects/4"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11.5 9C10.12 9 9 10.12 9 11.5s1.12 2.5 2.5 2.5 2.5-1.12 2.5-2.5S12.88 9 11.5 9zM20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm-3.21 14.21l-2.91-2.91c-.69.44-1.51.7-2.39.7C9.01 16 7 13.99 7 11.5S9.01 7 11.5 7 16 9.01 16 11.5c0 .88-.26 1.69-.7 2.39l2.91 2.9-1.42 1.42z"
                        />
                      </svg>
                    </a>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </li>
        <li
          class="MuiGridListTile-root"
          style="width: 66.66666666666667%; height: 350px; padding: 25px;"
        >
          <div
            class="MuiGridListTile-tile"
          >
            <img
              alt="I'll Server Soup - Kitchen Management App - Marketing Page"
              class="makeStyles-image-91"
              src="https://i.imgur.com/XvTo2Ma.png"
            />
            <div
              class="MuiGridListTileBar-root makeStyles-titleBar-89 MuiGridListTileBar-titlePositionTop"
            >
              <div
                class="MuiGridListTileBar-titleWrap MuiGridListTileBar-titleWrapActionPosLeft"
              >
                <div
                  class="MuiGridListTileBar-title"
                >
                  I'll Server Soup - Kitchen Management App - Marketing Page
                </div>
              </div>
              <div
                class="MuiGridListTileBar-actionIcon MuiGridListTileBar-actionIconActionPosLeft"
              >
                <button
                  aria-label="star I'll Server Soup - Kitchen Management App - Marketing Page"
                  class="MuiButtonBase-root MuiIconButton-root makeStyles-icon-90"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <a
                      href="/projects/5"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11.5 9C10.12 9 9 10.12 9 11.5s1.12 2.5 2.5 2.5 2.5-1.12 2.5-2.5S12.88 9 11.5 9zM20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm-3.21 14.21l-2.91-2.91c-.69.44-1.51.7-2.39.7C9.01 16 7 13.99 7 11.5S9.01 7 11.5 7 16 9.01 16 11.5c0 .88-.26 1.69-.7 2.39l2.91 2.9-1.42 1.42z"
                        />
                      </svg>
                    </a>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="makeStyles-root-120"
      id="test"
    >
      <div
        class="makeStyles-container-121"
      >
        <div
          class="makeStyles-root-122"
        >
          <h1>
            CONTACT
          </h1>
          <div>
            <a
              href="https://twitter.com/petrussola"
              rel="noopener noreferrer"
              target="_blank"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root makeStyles-icon-124"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </a>
            <a
              href="https://www.linkedin.com/in/peresola/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root makeStyles-icon-124"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </a>
            <a
              href="https://github.com/petrussola"
              rel="noopener noreferrer"
              target="_blank"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root makeStyles-icon-124"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </a>
          </div>
          <div
            class="makeStyles-root-125"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <defs>
                <path
                  d="M24 0v24H0V0h24z"
                  id="a"
                />
              </defs>
              <path
                d="M10.08 10.86c.05-.33.16-.62.3-.87s.34-.46.59-.62c.24-.15.54-.22.91-.23.23.01.44.05.63.13.2.09.38.21.52.36s.25.33.34.53.13.42.14.64h1.79c-.02-.47-.11-.9-.28-1.29s-.4-.73-.7-1.01-.66-.5-1.08-.66-.88-.23-1.39-.23c-.65 0-1.22.11-1.7.34s-.88.53-1.2.92-.56.84-.71 1.36S8 11.29 8 11.87v.27c0 .58.08 1.12.23 1.64s.39.97.71 1.35.72.69 1.2.91 1.05.34 1.7.34c.47 0 .91-.08 1.32-.23s.77-.36 1.08-.63.56-.58.74-.94.29-.74.3-1.15h-1.79c-.01.21-.06.4-.15.58s-.21.33-.36.46-.32.23-.52.3c-.19.07-.39.09-.6.1-.36-.01-.66-.08-.89-.23-.25-.16-.45-.37-.59-.62s-.25-.55-.3-.88-.08-.67-.08-1v-.27c0-.35.03-.68.08-1.01zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </svg>
            <div>
              2019 by Pere Solà
            </div>
          </div>
        </div>
        <div
          class="makeStyles-root-127"
        >
          <form
            action="#"
            class="makeStyles-form-128"
          >
            <input
              class="makeStyles-field-129"
              name="name"
              placeholder="Name"
              type="text"
              value=""
            />
            <input
              class="makeStyles-field-129"
              name="email"
              placeholder="Email"
              type="email"
              value=""
            />
            <textarea
              class="makeStyles-field-129"
              name="message"
              placeholder="Text"
            />
            <button
              class="makeStyles-field-129 makeStyles-submitButton-130 submitButtonFooter"
              type="submit"
            >
              SUBMIT
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
